src/components/Comments/Comment.js

```diff
@@ -169,6 +171,19 @@ const Comment = ({
     }
   };
 
+  const onPressDm = async () => {
+    try {
+      refSheet.current.open();
+      setLoading({...loading, loadingGetAllowAnonDmStatus: true});
+      const data = await getAllowAnonDmStatus('comment', comment?.id);
+      setUserAllowDm(data?.user.allow_anon_dm);
+    } catch (e) {
+      console.log(e);
+    } finally {
+      setLoading({...loading, loadingGetAllowAnonDmStatus: false});
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This new function `onPressDm` fetches the status of whether the user allows anonymous direct messages or not. It's good to see error handling and loading state management in place. However, the console log at line 181 should be replaced with proper error logging or reporting mechanism.
```diff
- console.log(e);
+ // Use your preferred error reporting service
+ reportError(e);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->